systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1069,6 +1077,453 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+
+        LOGGER.info("Recreate kafkaClients pod with new passwords.");
+        resourceManager.deleteResource(kubeClient().namespace(testStorage.getNamespaceName()).getDeployment(testStorage.getKafkaClientsName()));
+        resourceManager.createResource(extensionContext, KafkaClientsTemplates.kafkaClients(testStorage.getNamespaceName(), true, testStorage.getKafkaClientsName(), userSource, userTarget).build());
+        kafkaClientsPodName = kubeClient().listPodsByPrefixInName(testStorage.getKafkaClientsName()).get(0).getMetadata().getName();
+
+        internalKafkaClient = internalKafkaClient.toBuilder()
+            .withUsingPodName(kafkaClientsPodName)
+            .withTopicName(topicSourceNameB)
+            .withClusterName(kafkaClusterSourceName)
+            .withKafkaUsername(userSource.getMetadata().getName())
+            .withListenerName(kafkaTlsScramListenerName)
+            .build();
+        sent = internalKafkaClient.sendMessagesTls();
+
+        internalKafkaClient = internalKafkaClient.toBuilder()
+            .withTopicName(topicTargetNameB)
+            .withClusterName(kafkaClusterTargetName)
+            .withKafkaUsername(userTarget.getMetadata().getName())
+            .withConsumerGroupName(ClientUtils.generateRandomConsumerGroup())
+            .build();
+
+        LOGGER.info("Now messages should be mirrored to target topic and cluster");
+        internalKafkaClient.consumesTlsMessagesUntilOperationIsSuccessful(sent);
+        LOGGER.info("Messages successfully mirrored");
+    }
+
+    @ParallelNamespaceTest
+    @SuppressWarnings({"checkstyle:MethodLength"})
+    void testKMM2RollAfterSecretsCertsUpdateTLS(ExtensionContext extensionContext) {
+        final String namespaceName = StUtils.getNamespaceBasedOnRbac(INFRA_NAMESPACE, extensionContext);
+        final String kafkaClientsName = mapWithKafkaClientNames.get(extensionContext.getDisplayName());
+        String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        String kafkaClusterSourceName = clusterName + "-source";
+        String kafkaClusterTargetName = clusterName + "-target";
+        String topicSourceNameA = MIRRORMAKER2_TOPIC_NAME + "-" + rng.nextInt(Integer.MAX_VALUE);
+        String topicTargetNameA = kafkaClusterSourceName + "." + topicSourceNameA;
+        String topicSourceNameB = MIRRORMAKER2_TOPIC_NAME + "-" + rng.nextInt(Integer.MAX_VALUE);
+        String topicTargetNameB = kafkaClusterSourceName + "." + topicSourceNameB;
+        String kafkaUserSourceName = clusterName + "-my-user-source";
+        String kafkaUserTargetName = clusterName + "-my-user-target";
+
+        // Deploy source kafka with tls listener and mutual tls auth
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(kafkaClusterSourceName, 3, 3)
```

Comment:
In previous test you are using `kafkaPersistent` with 1 Kafka and 1 Zookeeper replicas. What's the difference between the two tests that you are using something different?

Reply:
TLS needs to use 3 ZK and K. So that's why this test uses 3 pairs.
Info provided by @sknot-rh - as he did fix.